zst/column/union.go

```diff
@@ -24,11 +25,18 @@ func NewUnionWriter(typ *zed.TypeUnion, spiller *Spiller) *UnionWriter {
 		typ:      typ,
 		values:   values,
 		selector: NewIntWriter(spiller),
+		presence: NewPresenceWriter(spiller),
 	}
 }
 
 func (u *UnionWriter) Write(body zcode.Bytes) error {
-	_, selector, zv := u.typ.SplitZNG(body)
+	if len(body) == 0 {
+		u.presence.TouchNull()
+		return nil
+	}
+	u.presence.TouchValue()
+	typ, zv := u.typ.SplitZNG(body)
+	selector := u.typ.Selector(typ)
```

Comment:
I think we should keep the `SplitZNG` selector return parameter so we don't have to do this lookup, but I don't feel strongly about that.